apps/aehttp/src/aehttp_helpers.erl

```diff
@@ -356,8 +356,13 @@ parse_map_to_atom_keys() ->
 poi_decode(PoIKey) ->
     fun(_Req, State) ->
         PoIBin = maps:get(PoIKey, State),
-        PoI = aec_trees:deserialize_poi(PoIBin),
-        {ok, maps:put(PoIKey, PoI, State)}
+        try aec_trees:deserialize_poi(PoIBin) of
+            PoI ->
+                {ok, maps:put(PoIKey, PoI, State)}
+        catch
+            _:_ ->
+                {error, {400, [], #{<<"reason">> => <<"Invalid proof of inclusion">>}}}
```

Comment:
Is the error in swagger spec, too?

Reply:
So this is used by the
* close solo and it has [code 400 handled](https://github.com/aeternity/epoch/blob/master/config/swagger.yaml#L1440)
* slash and it has [code 400 handled](https://github.com/aeternity/epoch/blob/master/config/swagger.yaml#L1465)

It will be used in the force progress once there is an endpoint for it as well :)